src/background/service/rpc.ts

```diff
@@ -0,0 +1,96 @@
+import { CHAINS_ENUM } from '@debank/common';
+import { createPersistStore } from 'background/utils';
+import axios from 'axios';
+
+export type RPCServiceStore = {
+  customRPC: Record<string, string>;
+};
+
+const MAX = 4_294_967_295;
+let idCounter = Math.floor(Math.random() * MAX);
+
+function getUniqueId(): number {
+  idCounter = (idCounter + 1) % MAX;
+  return idCounter;
+}
+
+class RPCService {
+  store: RPCServiceStore = {
+    customRPC: {},
+  };
+
+  init = async () => {
+    const storage = await createPersistStore<RPCServiceStore>({
+      name: 'rpc',
+      template: {
+        customRPC: {},
+      },
+    });
+    this.store = storage || this.store;
+  };
+
+  hasCustomRPC = (chain: CHAINS_ENUM) => {
+    return !!this.store.customRPC[chain];
+  };
+
+  getRPCByChain = (chain: CHAINS_ENUM) => {
+    return this.store.customRPC[chain];
+  };
+
+  getAllRPC = () => {
+    return this.store.customRPC;
+  };
+
+  setRPC = (chain: CHAINS_ENUM, url: string) => {
+    this.store.customRPC = {
+      ...this.store.customRPC,
+      [chain]: url,
+    };
+  };
+
+  removeCustomRPC = (chain: CHAINS_ENUM) => {
+    const map = this.store.customRPC;
+    delete map[chain];
+    this.store.customRPC = map;
+  };
+
+  requestCustomRPC = async (
+    chain: CHAINS_ENUM,
+    method: string,
+    params: any[]
+  ) => {
+    const host = this.store.customRPC[chain];
+    if (!host) {
+      throw new Error(`No customRPC set for ${chain}`);
+    }
+    return this.request(host, method, params);
+  };
+
+  request = async (
+    host: string,
+    method: string,
+    params: any[],
+    timeout?: number
+  ) => {
+    const { data } = await axios.post(
+      host,
+      {
+        jsonrpc: '2.0',
+        id: getUniqueId(),
```

Comment:
For loaded background, return value of `getUniqueId` is static here.

Maybe it's better to it set `id` as different value on every call of `RPCService::request`?

Reply:
> it's feature, when background loaded, `idCounter` will init as a random number and everytime we call `getUniqueId`, `idCounter` will increase

My missing, I didn't see the line `idCounter = (idCounter + 1) % MAX;` 🤣

Reply:
> it's feature, when background loaded, `idCounter` will init as a random number and everytime we call `getUniqueId`, `idCounter` will increase

> it's feature, when background loaded, `idCounter` will init as a random number and everytime we call `getUniqueId`, `idCounter` will increase

My missing, I didn't see the line `idCounter = (idCounter + 1) % MAX;` 🤣